### DISCUSSION TITLE
Show 3D Tiles attributes

### DISCUSSION THREAD

**Discussion by imakihi** [TerriaJS/terriajs#5064] [Q&A] [answered]
Hi there. I have another question regarding showing 3D Tiles attributes. 

My 3D Tiles data have building names and their heights as their attributes. How I can show these attributes on Terria.io? I looked at the featureInfoTemplate specifications and some examples. It looks like I need a separate json file to store attribute data to use this option? Can I show attributes stored in my 3D Tiles data directly?

When I used Cesium.js, I could show those attributes but I don't know how to do it with Terria.io.

Can anyone kindly point a right direction? Thank you in advance.

Hiroo

**Comment by KeyboardSounds** [comment]
Hey @imakihi,

Have you tried loading the 3d tiles in Terria? It should 🌟 just work 🌟— if you click in a feature in the tileset, you should see the attributes for that feature pop up in the top right of the window. We do support putting attributes in a separate JSON file, but it's not necessary if they're already embedded in the tileset.
Are you able to share your 3d tiles dataset with us? Then I can have a look and see what's happening.

**Reply by meh9** [reply]
@imakihi 
e.g. here is a 3DTiles photomesh that has leftover `id` attributes: https://nsw.digitaltwin.terria.io/#share=s-lWaM3t4gNWuCiXNlGtVz9vILKY7
and here is an i3s-to-3dtiles converted dataset with lots of attributes when you click: https://nsw.digitaltwin.terria.io/#share=s-hSqhEeHqEXJn55Sx8oB9BVtke9r

**Reply by hardreddata** [reply]
I understand this only works on the _Next_ branch and not the current master TerriaMap

**Reply by KeyboardSounds** [reply]
@RussellGrew which part? i3s-to-3dtiles hasn't been released to the general public, ~but picking attributes in 3d tilesets should work in v7.~ or not, see @meh9's answer 😬

**Reply by imakihi** [reply]
@KeyboardSounds @meh9 @RussellGrew 
Thank you so much for helping. I am currently using the current master TerriaMap same as @RussellGrew. I could show 3D Tiles data but I cannot show the attributes when I click building I loaded. Do I need to do something special to work the info window function on V7?

Again, thank you so much!!!
<img width="706" alt="2020-12-08_16-39-43" src="https://user-images.githubusercontent.com/4824436/101558590-23876f00-3974-11eb-8cad-a839f8a4c665.png">

**Reply by hardreddata** [reply]
From @soyarsauce a while back in the gitter, when I was pining for the coloured buildings

![image](https://user-images.githubusercontent.com/42530666/101559524-66e5ea00-3a15-11eb-86b2-ef0ca2156558.png)

Assuming that styles and attributes are dealt with in the same bit of code.

**Comment by meh9** [comment, marked as answer]
@RussellGrew @imakihi aha, we've looked into it and it looks like feature picking was never incorporated in TerriaJS v7 - i.e. current master. There is a branch that has it for v7 here that you might be able to use: https://github.com/TerriaJS/terriajs/compare/3d-tiles-styling

However, as we are not working on v7 any more, our very strong recommendation is that you use v8 for all new work.

**Comment by imakihi** [comment]
@meh9 It sounds like I need to move to V8. 
Thank you for making sure that the feature doesn't work on the current V7 master.

Best regards.

**Comment by imakihi** [comment]
Hi all. I just updated from V7 to V8 and I can show attributes without any issues.
Thank you for your help!
<img width="583" alt="2020-12-09_16-54-58" src="https://user-images.githubusercontent.com/4824436/101707252-b80ad380-3a3f-11eb-961f-af8a6702a1f0.png">
